version: '3'

services:
  jupyterhub:
    build: jupyterhub
    container_name: jupyterhub
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - jupyterhub_data:/srv/jupyterhub
    environment:
      DOCKER_JUPYTER_CONTAINER: root_jupyterlab
      DOCKER_NETWORK_NAME: root_default
      HUB_IP: jupyterhub
    labels:
      - "traefik.enable=true"
      - "traefik.hub.frontend.rule=Host:jupyter.ens.uvsq.fr"
    restart: on-failure
  jupyterlab:
    build: jupyterlab
    container_name: jupyterlab-throaway
    command: echo
  reverse-proxy:
    image: traefik
    container_name: reverse-proxy
    ports:
      - "80:80"
      - "443:443"
      - "8080:8080"
    volumes:
      - ./reverse-proxy/traefik.toml:/etc/traefik/traefik.toml
      - /etc/certs:/etc/certs
      - /var/run/docker.sock:/var/run/docker.sock
    restart: on-failure

volumes:
  jupyterhub_data:
